src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/metadata/ProjectSampleMetadataController.java

```diff
@@ -60,13 +64,13 @@ public ProjectSampleMetadataController(MessageSource messageSource, ProjectServi
 	}
 
 	/**
-	 * Upload Excel file containing sample metadata and extract the headers.  The file is stored in the session until
+	 * Upload CSV or Excel file containing sample metadata and extract the headers.  The file is stored in the session until
 	 * the column that corresponds to a {@link Sample} identifier has been sent.
 	 *
 	 * @param session   {@link HttpSession}
 	 * @param projectId {@link Long} identifier for the current {@link Project}
-	 * @param file      {@link MultipartFile} The excel file containing the metadata.
-	 * @return {@link Map} of headers and rows from the excel file for the user to select the header corresponding the
+	 * @param file      {@link MultipartFile} The csv or excel file containing the metadata.
+	 * @return {@link Map} of headers and rows from the csv or excel file for the user to select the header corresponding the
 	 * {@link Sample} identifier.
 	 */
 	@RequestMapping(value = "/upload/file", method = RequestMethod.POST)
```

Comment:
Can you move most of the contents of this out into a service class (maybe `UIMetadataImportService`)? This method is huge (I know because I wrote the thing :)).  An maybe extract out the csv and xslx into their own functions?  I really want to streamline the controllers as much as possible.

Reply:
Is this closer to what you're thinking??? Please see e56d228e3ad71665fb73fb852fcf7ee3fc15f5f5 and 7266c128efcff57f39e5058ca9f635af7abca1ec.

Reply:
This is exactly what I was looking for!  Thank you.